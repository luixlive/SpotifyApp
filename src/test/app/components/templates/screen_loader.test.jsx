import _ from 'lodash';
import React from 'react';
import renderer from 'react-test-renderer';

import {
  screenLoader as componentProps,
} from './../../../test_utils/components_props';
import ScreenLoader from './../../../../app/components/templates/screen_loader';

describe('App Components Templates - Screen Loader', () => {
  describe('Snapshots', () => {
    let props;
    beforeEach(() => {
      props = _.cloneDeep(componentProps);
    });

    it('renders', () => {
      const rendered = renderer.create(<ScreenLoader {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders with custom text', () => {
      props.text = 'text';
      const rendered = renderer.create(<ScreenLoader {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });
  });
});
