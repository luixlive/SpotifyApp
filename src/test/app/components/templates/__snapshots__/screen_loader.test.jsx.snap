// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components Templates - Screen Loader Snapshots renders 1`] = `
<div
  className="ui basic segment screen-loader"
>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="center"
      >
        <div
          className="ui big inverted text loader"
        >
          <h3
            className="ui header"
          >
            
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App Components Templates - Screen Loader Snapshots renders with custom text 1`] = `
<div
  className="ui basic segment screen-loader"
>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="center"
      >
        <div
          className="ui big inverted text loader"
        >
          <h3
            className="ui header"
          >
            text
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;
